src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -728,11 +728,12 @@ EnsureRelationCanBeDistributed(Oid relationId, Var *distributionColumn,
 		}
 
 		/* we currently don't support MX tables to be distributed partitioned table */
-		if (replicationModel == REPLICATION_MODEL_STREAMING)
+		if (replicationModel == REPLICATION_MODEL_STREAMING &&
+			CountPrimariesWithMetadata() > 0)
```

Comment:
I'm not suggesting we do it here (or that's the best API), but to understand the function, can we write the following:

```python
IsMxCluster():
    if CountPrimariesWithMetadata() >0:
       return true
   return false
```

Reply:
It kind of makes sense, but I would expect `IsMxCluster` to tell me whether I'm in an MX cluster on the worker nodes as well, but it won't, since the hasmetadata flag is not reliably set on the workers.